swap_meet/vendor.py

```diff
@@ -1,2 +1,91 @@
 class Vendor:
-    pass
\ No newline at end of file
+    def __init__(self, inventory=None):
+        if not inventory:
+            self.inventory = []
+        else:
+            self.inventory = inventory
+
+    def add(self, item):
+        self.inventory.append(item)
+        return item
+
+    def remove(self, item):
+        if item in self.inventory:
+            self.inventory.remove(item)
+            return item
+
+    def get_by_id(self, item_id):
+        for item in self.inventory:
+            if item.id == item_id:
+                return item
+        
+    def swap_items(self, other_vendor, my_item, their_item):
+        if my_item not in self.inventory or their_item not in other_vendor.inventory:
+            return False
+        other_vendor.inventory.remove(their_item)
+        self.inventory.append(their_item)
+        self.inventory.remove(my_item)
+        other_vendor.inventory.append(my_item)
+        return True
+    
+    def swap_first_item(self, other_vendor):
+        if not self.inventory or not other_vendor.inventory:
+            return False
+        first_item = self.inventory[0]
+        first_item_other = other_vendor.inventory[0]
+        if  first_item  and first_item_other:
+            self.inventory.remove(first_item)
+            other_vendor.inventory.append(first_item)
+            other_vendor.inventory.remove(first_item_other)
+            self.inventory.append(first_item_other)
+            return True
+    
+    def get_by_category(self, category="Unknown"):
+        return[item for item in self.inventory if item.get_category() == category]
+    
+    def get_highest_item(self, list_items, key=lambda x: x):
+        if not list_items:
+            return False
+        
+        highest = list_items[0]
+        for item in list_items:
+            if key(item) > key(highest):
+                highest = item
+        return highest
+    
+    def get_best_by_category(self, category):
+        items_in_category = self.get_by_category(category)
+
+        if not items_in_category:
+            return False
+
+        best_item = self.get_highest_item(items_in_category, key= lambda item: item.condition)
```

Comment:
Nice job writing a function that takes a lambda expression! It's good to see you getting practice with passing functions to another function since you couldn't use `max` for the project. 🎉

In the future and in practice, you would just use `max` or write a for loop to get the highest item instead of re-writing max's logic like you did (but again this is good for getting familiar with how the Python code works under the hood).

Reply:
Nit: extra whitespace
```suggestion
        best_item = self.get_highest_item(items_in_category, key=lambda item: item.condition)
```

Reply:
Consider extracting `get_highest_item` into a helper function that is in its own file (maybe something like `item_helpers.py`) instead of creating it as an instance method here. It feels a little funny that the `Vendor` class has the responsibility of finding the highest item so putting it in a helper would remove this logic from `Vendor`.

Same comment applies for `get_newest_item`.

Reply:
Will ask about this later

Reply:
We think we fixed this!